distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerProcessTests.java

```diff
@@ -373,7 +373,7 @@ public void testWaitFor() throws Exception {
         new Thread(() -> {
             // simulate stop run as shutdown hook in another thread, eg from Ctrl-C
             nonInterruptibleVoid(mainReady::await);
-            server.stop();
+            IOUtils.closeWhileHandlingException(server::stop);
```

Comment:
If stop throws shouldn't we fail the test? Why do we hide the exception?

Reply:
Good point, there's in deed a race between `server.stop()` and `server.waitFor()`. 
IOExceptions of the `ErrorPumpThread` can bubble up on either one. I'll fix 👍